### DISCUSSION TITLE
[Feature] Support Azure OpenAI Service

### DISCUSSION THREAD

**Discussion by jhhuang96** [immersive-translate/immersive-translate#716] [Ideas]
### 新功能 | New Feature

Support Azure OpenAI

### 为什么您想要添加这些功能 | Why you want to add these features

Azure OpenAI Service provides REST API access to OpenAI's powerful language models including the GPT-3, Codex and Embeddings model series. ([link](https://learn.microsoft.com/en-us/azure/cognitive-services/openai/overview))

Azure OpenAI Service gives customers advanced language AI with OpenAI GPT-4, GPT-3, Codex, and DALL-E models with the security and enterprise promise of Azure. Azure OpenAI co-develops the APIs with OpenAI, ensuring compatibility and a smooth transition from one to the other.

With Azure OpenAI, customers get the security capabilities of Microsoft Azure while running the same models as OpenAI. Azure OpenAI offers private networking, regional availability, and responsible AI content filtering.

Reference: https://learn.microsoft.com/en-us/azure/cognitive-services/openai/reference

### 提议 | Proposal

Sample commit: https://github.com/openai-translator/openai-translator/commit/530467843de8b649e0f0865096eba51bf18953f6

**Comment by theowenyoung** [comment]
已经支持的哈，参见文档：

https://immersivetranslate.com/docs/services/openai/#azure-openai
